soco/exceptions.py

```diff
@@ -80,3 +80,60 @@ class SoCoSlaveException(SoCoException):
 
 class NotSupportedException(SoCoException):
     """Raised when something is not supported by the device"""
+
+
+class EventParseException(SoCoException):
+    """Raised when a parsing exception occurs during event handling."""
+
```

Comment:
As with the other one, I think i would be a good idea to document the instance attributes. From the docs: http://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html it seems that the way to do it is to add this section to the class docstring:

```
Attributes:
    tag (str): The tag for which the exception occured
    metadata (str): The metadata which failed to parse
    __cause__ (Exception): The original exception
```

Why is cause marked with double underscores btw?

Reply:
`__cause__` is a special Exception attribute for chaining. The output looks like this:

```python
Traceback (most recent call last):
  File "C:/Users/Christian/.PyCharmCE2017.3/config/scratches/scratch_2.py", line 5, in <module>
    raise SoCoException('my_message')
soco.exceptions.SoCoException: my_message

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:/Users/Christian/.PyCharmCE2017.3/config/scratches/scratch_2.py", line 13, in <module>
    value.xyz
  File "C:\Users\Christian\PycharmProjects\SoCo\soco\exceptions.py", line 121, in __getattr__
    raise self.exception
soco.exceptions.EventParseException: Invalid metadata for 'my_tag'
```

Reply:
Should I include `__cause__` in the attributes section anyways?

Reply:
Oh wow. I tried googling that and came up empty. Must have been weak google-foo. I'm unsure if that works in Python 2.7 also, but I guess if it doesn't, then it just wont do anything.